_inc/client/setup-wizard/feature-toggle/index.jsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * External dependencies
+ */
+import { Disabled, FormToggle } from '@wordpress/components';
+import classnames from 'classnames';
+import { translate as __ } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Gridicon from 'components/gridicon';
+
+import {
+	mapStateToFeatureToggleProps,
+	mapDispatchToFeatureToggleProps,
+} from './map-feature-to-props';
+
+import './style.scss';
+
+let FeatureToggle = props => {
+	const {
+		title,
+		details,
+		info,
+		checked,
+		configureLink,
+		upgradeLink,
+		optionsLink,
+		isPaid = false,
+		isLinkExternal = false,
+	} = props;
+
+	const [ windowWidth, setWindowWidth ] = useState( false );
+
+	const handleResize = useCallback( () => {
+		setWindowWidth( window.innerWidth <= 660 ? 'small' : 'large' );
+	}, [ window.innerWidth ] );
+
+	useEffect( () => {
+		handleResize(); // Call this once to make sure windowWidth is initialized
+		window.addEventListener( 'resize', handleResize );
+		return () => {
+			window.removeEventListener( 'resize', handleResize );
+		};
+	} );
+
+	const onToggleChange = useCallback( () => {
+		if ( 'function' === typeof props.onToggleChange ) {
+			props.onToggleChange( checked );
+		}
+	}, [ checked, props.onToggleChange ] );
+
+	let buttonContent;
+	if ( ! checked && upgradeLink ) {
+		buttonContent = (
+			<Button href={ upgradeLink } primary target={ isLinkExternal ? '_blank' : '' }>
+				{ __( 'Upgrade now' ) }
+				{ isLinkExternal && (
+					<span>
+						<Gridicon icon="external" />
+					</span>
+				) }
+			</Button>
+		);
+	} else if ( configureLink ) {
+		buttonContent = (
+			<Button href={ configureLink } target={ isLinkExternal ? '_blank' : '' }>
+				{ __( 'Configure' ) }
+				{ isLinkExternal && (
+					<span>
+						<Gridicon icon="external" />
+					</span>
+				) }
+			</Button>
+		);
+	}
+
+	const largeWindow = 'large' === windowWidth;
+	const smallWindow = 'small' === windowWidth;
+
+	let infoContent;
+	if ( info ) {
+		infoContent = <p className="jp-setup-wizard-feature-toggle-info">{ info }</p>;
+	}
+
+	let optionsLinkElement;
+	if ( optionsLink ) {
+		optionsLinkElement = (
+			<a
+				href={ optionsLink }
+				target="_blank"
+				rel="noopener noreferrer"
```

Comment:
This really surprised me when testing. I would not expect a new window to open for something that's in the same dashboard.

Reply:
@robertf4 I wonder if this was left behind and should be removed.

Reply:
I updated this so that external links open in a new tab and have an external icon, while dashboard links do not have an icon and stay in the same tab.